src/pages/UserProfile.tsx

```diff
@@ -0,0 +1,192 @@
+import { Notification } from '../components/Notification/Notification';
+import type { JSXElement } from '../types/jsxElement';
+import './UserProfile.less';
+import Institutions from './UserProfileInstitutions.json';
+
+const emailOptions = [
+  '',
+  'test.user@bank0.com',
+  'test.user@financial_institution_email_domain-not-in-system.com',
+  'has-multiple-institutions@foo.bar'
+];
+
+const filterInstitutionOptionsByDomain = emailDomain => {
+  /**
+   * Example Institution
+   {
+    "name": "Test Bank 230626",
+    "lei": "TESTBANK230626",
+    "domains": [
+      {
+        "domain": "baz.qux",
+        "lei": "TESTBANK230626"
+      },
+      {
+        "domain": "foo.bar",
+        "lei": "TESTBANK230626"
+      }
+    ]
+  }
+   */
+  const matches = Institutions.filter(index => {
+    if (index.domains.length === 0 > 0) return false;
+    const _matches = index.domains.filter(
+      ({ domain }) => domain === emailDomain
+    );
+    return _matches.length > 0;
+  });
+
+  console.log('matches', matches);
+  console.log(
+    'institution matches',
+    matches.map(match => makeSelectOption(match.name, match.lei))
+  );
+  return matches.map(match =>
+    makeSelectOption(`${match.name} | ${match.lei}`, match.lei)
+  );
+};
+
+export default function UserProfile({
+  emailAddress = emailOptions[0],
+  firstName = 'Test',
+  lastName = 'User'
+}): JSX.Element {
+  const [email, setEmail] = useState(emailAddress);
+  const [institutions, setInstitutions] = useState([]);
+
+  const isValidEmailDomain = email.length > 0;
+
+  useEffect(() => {
+    setInstitutions([]);
+  }, [email]);
+
+  const onEmailChange = useCallback(
+    event => {
+      console.log('Changing email to', event.target.value);
+      setEmail(event.target.value);
+    },
+    [setEmail]
+  );
+
+  const selectableInstitutions = useMemo(
+    () => filterInstitutionOptionsByDomain(email.split('@')[1]),
+    [email]
+  );
+
+  const hasMatchingInstitutions = selectableInstitutions.length > 0;
+
+  return (
+    <main className='page'>
+      <h1>Complete your user profile</h1>
+      <p className='intro'>
+        Lorem ipsum dolor sit amet consectetur adipisicing elit. Earum commodi
+        atque ut quidem ipsum! Nemo inventore magnam dolorem, nihil cumque
+        explicabo qui quisquam sit tempore quam, voluptate natus officia
+        doloremque.
+      </p>
+      <Notification
+        message='Does your financial institution have an LEI?'
+        type='warning'
+      >
+        In order to begin using the filing platform your institution must have a
+        Legal Entity Identifier (LEI). Visit the Global LEI Foundation (GLEIF)
+        website for more information on how to obtain an LEI for your
+        institution.
+      </Notification>
+      <div className='form'>
+        <form action=''>
+          <div className='field'>
+            <h3>First name</h3>
+            <input type='text' defaultValue={firstName} />
```

Comment:
I know this PR is WIP but it should be straightforward to swap in our `TextInput`:

```suggestion
            <TextInput
              type='text'
              id='firstName'
              name='firstName'
              defaultValue={firstName}
            />
```